src/main/java/software/amazon/awssdk/http/apache/internal/conn/SdkTlsSocketFactory.java

```diff
@@ -71,16 +82,22 @@ public Socket connectSocket(
             final HttpHost host,
             final InetSocketAddress remoteAddress,
             final InetSocketAddress localAddress,
-            final HttpContext context) throws IOException {
-        log.trace(() -> String.format("Connecting to %s:%s", remoteAddress.getAddress(), remoteAddress.getPort()));
+            final HttpContext context)
+            throws IOException {
+        log.trace(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code logs sensitive information (such as PAN numbers, credit cards, or access tokens) which could expose confidential data to unauthorized personnel through system logs and monitoring tools. Such exposure could lead to financial fraud, identity theft, or regulatory violations. Consider using data masking or removing sensitive information from log statements to maintain data security.

[Learn more](https://cwe.mitre.org/data/definitions/200.html)